client/packages/system/src/Report/components/ReportArgumentsModal.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC, useState } from 'react';
+
+import { JsonData, JsonForm } from '@openmsupply-client/programs';
+import { ReportRowFragment } from '../api';
+import { useDialog } from '@common/hooks';
+import { DialogButton } from '@common/components';
+
+export type ReportArgumentsModalProps = {
+  /** Modal is shown if there is an argument schema present */
+  report: ReportRowFragment | undefined;
+  onReset: () => void;
+  onArgumentsSelected: (report: ReportRowFragment, args: JsonData) => void;
+};
+
+export const ReportArgumentsModal: FC<ReportArgumentsModalProps> = ({
+  report,
+  onReset,
+  onArgumentsSelected,
+}) => {
+  const [data, setData] = useState<JsonData>({});
+  const [error, setError] = useState<string | false>(false);
+
+  // clean up when modal is closed
+  const cleanUp = () => {
+    setData({});
+    onReset();
+  };
+
+  const { Modal } = useDialog({
+    isOpen: !!report?.argumentSchema,
+    onClose: cleanUp,
+  });
+
+  if (!report?.argumentSchema) {
+    return null;
+  }
+
+  return (
+    <Modal
+      title="Report arguments"
+      cancelButton={<DialogButton variant="cancel" onClick={cleanUp} />}
+      okButton={
```

Comment:
I think it is better with `slideAnimation={false}` ( the slide animation is primarily for ok/next modals )

Reply:
thanks done